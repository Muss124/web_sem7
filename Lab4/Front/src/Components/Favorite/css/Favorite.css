.FavoriteContainer {
    width          : 99%;
    display        : flex;
    flex-wrap      : wrap;
    flex-direction : row;
    justify-content: space-around;
}