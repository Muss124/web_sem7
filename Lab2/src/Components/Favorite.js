import React from 'react';
import { FavoriteSearch } from "./FavoriteSearch.js"
import { connect } from 'react-redux'
import { FavoriteItem } from "./FavoriteItem.js"

class Favorite extends React.Component {
    render() {
        return (
            <div>
                <FavoriteSearch />
                {
                    this.props.data.map((favCity, i = 0) => {
                        i++;
                        return ([
                            <div class="FavoriteContainer" key={i}>
                                <FavoriteItem value={favCity}  />
                            </div>
                        ])
                    })
                }

            </div>
        )
    }
}

const mapStateToProps = (state) => {
    console.log(state);
    return {
        data: state
    };
}

const mapDispatchToProps = (dispatch) => {
    return {
        setName: () => {
            dispatch({
            });
        }
    };
}
export default connect(mapStateToProps, mapDispatchToProps)(Favorite);