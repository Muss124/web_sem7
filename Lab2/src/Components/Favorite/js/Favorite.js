import React from 'react';
import { FavoriteSearch } from "../../FavoriteSearch/js/FavoriteSearch"
import { connect } from 'react-redux'
import { FavoriteItem } from "../../FavoriteItem/js/FavoriteItem"
import '../css/Favorite.css'

class Favorite extends React.Component {
    render() {
        return (
            <div>
                <FavoriteSearch />
                <div className="FavoriteContainer" >
                    {
                        this.props.data.map((favCity, i = 0) => {
                            i++;
                            return ([

                                <FavoriteItem value={favCity} key={i}/>

                            ])
                        })
                    }
                </div>
            </div>
        )
    }
}

const mapStateToProps = (state) => {
    console.log(state);
    return {
        data: state
    };
}

const mapDispatchToProps = (dispatch) => {
    return {
        setName: () => {
            dispatch({
            });
        }
    };
}
export default connect(mapStateToProps, mapDispatchToProps)(Favorite);