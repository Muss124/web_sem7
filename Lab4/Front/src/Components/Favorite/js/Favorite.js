import React from 'react';
import { FavoriteSearch } from "../../FavoriteSearch/js/FavoriteSearch"
import { connect } from 'react-redux'
import { FavoriteItem } from "../../FavoriteItem/js/FavoriteItem"
import '../css/Favorite.css'
import FetchFavorite from "../../../Actions/FetchFavorite"
import store from "../../../store"

class Favorite extends React.Component {
    constructor(props) {
        super(props);
        store.dispatch(FetchFavorite());
    }
    render() {
        return (
            <div>
                <FavoriteSearch />
                <div className="FavoriteContainer" >
                    {
                        this.props.data.map((favCity, i = 0) => {
                            i++;
                            return ([

                                <FavoriteItem value={favCity} key={i} />

                            ])
                        })
                    }
                </div>
            </div>
        )
    }
}

const mapStateToProps = (state) => {
    console.log(state);
    return {
        data: state
    };
}

const mapDispatchToProps = (dispatch) => {
    return {
        setName: () => {
            dispatch({
            });
        }
    };
}
export default connect(mapStateToProps, mapDispatchToProps)(Favorite);