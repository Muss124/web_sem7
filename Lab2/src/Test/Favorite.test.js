import React from 'react';
import Favorite from '../Components/Favorite/js/Favorite'
import { shallow } from '../enzyme';
import { shallowToJson } from 'enzyme-to-json';
import configureStore from 'redux-mock-store';

const initialState = {}
const mockStore = configureStore()

describe('Testing render Favorite', function () {
    it("Render from empty storage", () => {
        const storage = [];
        const store = mockStore(initialState);
        const tree = shallow(<Favorite value={storage} store={store} />);
        expect(shallowToJson(tree)).toMatchSnapshot();
    });

    it("Render from with loaded element", () => {
        const storage = [{
            "city": "Kiev",
            "data": {
                CityName: "Kiev",
                Temperature: 2.98,
                Weather: "light intensity drizzle",
                Humidity: 93,
                Pressure: 1014,
                Wind: 4,
                Longitude: 30.52,
                Latitude: 50.43,
                Icon: "09n",
                Cod: 200
            }
        }];
        const store = mockStore(initialState);
        const tree = shallow(<Favorite value={storage} store={store} />);
        expect(shallowToJson(tree)).toMatchSnapshot();
    });
    it("Render from empty with loading element", () => {
        const storage = [{
            "loading": true,
            "city": "TEST LOADING CITY",
            "data": {}
        }];
        const store = mockStore(initialState);
        const tree = shallow(<Favorite value={storage} store={store} />);
        expect(shallowToJson(tree)).toMatchSnapshot();
    });
    it("Render from empty with both loading and loaded element", () => {
        const storage = [
            {
                "city": "Kiev",
                "data": {
                    CityName: "Kiev",
                    Temperature: 2.98,
                    Weather: "light intensity drizzle",
                    Humidity: 93,
                    Pressure: 1014,
                    Wind: 4,
                    Longitude: 30.52,
                    Latitude: 50.43,
                    Icon: "09n",
                    Cod: 200
                }
            },
            {
                "loading": true,
                "city": "TEST LOADING CITY",
                "data": {}
            }
        ];
        const store = mockStore(initialState);
        const tree = shallow(<Favorite value={storage} store={store} />);
        expect(shallowToJson(tree)).toMatchSnapshot();
    });

});
