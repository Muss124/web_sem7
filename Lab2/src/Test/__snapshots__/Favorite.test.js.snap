// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing render Favorite Render from empty storage 1`] = `
<ContextProvider
  value={null}
>
  <Favorite
    data={Object {}}
    setName={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    value={Array []}
  />
</ContextProvider>
`;

exports[`Testing render Favorite Render from empty with both loading and loaded element 1`] = `
<ContextProvider
  value={null}
>
  <Favorite
    data={Object {}}
    setName={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    value={
      Array [
        Object {
          "city": "Kiev",
          "data": Object {
            "CityName": "Kiev",
            "Cod": 200,
            "Humidity": 93,
            "Icon": "09n",
            "Latitude": 50.43,
            "Longitude": 30.52,
            "Pressure": 1014,
            "Temperature": 2.98,
            "Weather": "light intensity drizzle",
            "Wind": 4,
          },
        },
        Object {
          "city": "TEST LOADING CITY",
          "data": Object {},
          "loading": true,
        },
      ]
    }
  />
</ContextProvider>
`;

exports[`Testing render Favorite Render from empty with loading element 1`] = `
<ContextProvider
  value={null}
>
  <Favorite
    data={Object {}}
    setName={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    value={
      Array [
        Object {
          "city": "TEST LOADING CITY",
          "data": Object {},
          "loading": true,
        },
      ]
    }
  />
</ContextProvider>
`;

exports[`Testing render Favorite Render from with loaded element 1`] = `
<ContextProvider
  value={null}
>
  <Favorite
    data={Object {}}
    setName={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    value={
      Array [
        Object {
          "city": "Kiev",
          "data": Object {
            "CityName": "Kiev",
            "Cod": 200,
            "Humidity": 93,
            "Icon": "09n",
            "Latitude": 50.43,
            "Longitude": 30.52,
            "Pressure": 1014,
            "Temperature": 2.98,
            "Weather": "light intensity drizzle",
            "Wind": 4,
          },
        },
      ]
    }
  />
</ContextProvider>
`;
